src/Middleware/SubstituteBindings.php

```diff
@@ -47,45 +49,91 @@ public function handle($request, Closure $next)
 
         $this->router->substituteBindings($route);
 
-        $this->substituteImplicitBindings($route);
+        //$this->substituteImplicitBindings($route);
+
+        $this->resolveForRoute($route);
 
         return $next($request);
     }
 
     /**
-     * Substitute the implicit Doctrine entity bindings for the route.
+     * Resolve the implicit route bindings for the given route.
      *
-     * @param Route $route
+     * @param  \Illuminate\Routing\Route  $route
+     * @return void
      *
-     * @throws EntityNotFoundException
+     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException
      */
-    protected function substituteImplicitBindings(Route $route)
+    public function resolveForRoute($route)
     {
         $parameters = $route->parameters();
 
         foreach ($this->signatureParameters($route) as $parameter) {
-            $id    = $parameters[$parameter->name];
-            $class = $parameter->getClass()->getName();
+            if (! $parameterName = static::getParameterName($parameter->name, $parameters)) {
+                continue;
+            }
 
-            if ($repository = $this->registry->getRepository($class)) {
-                if ($parameter->getClass()->implementsInterface(UrlRoutable::class)) {
-                    $name = call_user_func([$class, 'getRouteKeyName']);
+            $parameterValue = $parameters[$parameterName];
 
-                    $entity = $repository->findOneBy([
-                        $name => $id
-                    ]);
-                } else {
-                    $entity = $repository->find($id);
-                }
```

Comment:
I'm wondering if this whole block should appear at line 104 of the new changes, instead of just `$entity = $repository->find($parameterValue);`. I think this would mean the `getRouteKeyName` logic only applies if a custom key hasn't been defined as part of the route parameters (e.g. `{user:username}`).

Reply:
i think it should